lua/modules/hierarchyactions.lua

```diff
@@ -49,6 +49,11 @@ local hierarchyActionsMetatable = {
 				depth = newDepth,
 				includeRoot = true, -- always include root for descendants
 			}
+			if shape.ChildrenCount == nil then
```

Comment:
All Objects have a `ChildrenCount` field. 
But you're right, something should be done to better control `applyToDescendants` parameters upstream. 
My suggestion: 

```lua
applyToDescendants = function(self, shape, options, callback)
```

- `shape` parameter could be called `object` since it can technically be any kind of Object
- if `object` is not an Object, then we should raise an error like `error("applyToDescendants(object, options, callback) - object should be an Object, 2)` (`2` is to trigger error at the caller level)

Now to check if something is an Object, I think it's about time we introduce an `IsObject` function or something, to avoid testing against `Object`, `Shape`, `MutableShape`, `Quad` and future types...

@ArthurVoxowl what do you think about that last point?